test/unit/Container.spec.ts

```diff
@@ -128,13 +112,13 @@ describe("Container", function() {
             }
         }
 
-        function createCar (engine: Engine) {
-            return new Car(engine);
+        class CarFactory {
+            @Factory()
+            public static createCar (engine: Engine): Car {
```

Comment:
will reflect metadata work if you remove `: Car` from the end of this line? I guess no. And if no, then this method is not good - its hidden and unclear for a user to put return type explicitly when he have a type inference.